sklearn/mixture/tests/test_gaussian_mixture.py

```diff
@@ -569,6 +570,26 @@ def test_gaussian_mixture_predict_predict_proba():
         assert_greater(adjusted_rand_score(Y, Y_pred), .95)
 
 
+def test_gaussian_mixture_fit_predict():
+    rng = np.random.RandomState(0)
+    rand_data = RandomData(rng)
+    for covar_type in COVARIANCE_TYPE:
+        X = rand_data.X[covar_type]
+        Y = rand_data.Y
+        g = GaussianMixture(n_components=rand_data.n_components,
+                            random_state=rng, weights_init=rand_data.weights,
```

Comment:
I think we should be passing random_state=0 rather than passing an object which will be changed with each iteration

Reply:
I think it doesn't matter because the only thing we need is that the GMM is not changed within one iteration. Using different random_state with different COVARIANCE_TYPE actually tests more robustness? I guess?